### PR TITLE
Iteration1

### PR DESCRIPTION
**Pull request by alinazaitseva1**

### CODE REVIEW

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import UIKit
+
+class DataStructureTableViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    
+    let dataStructureManipulatios = DataCaprureMemoryStructModel()
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+//    var memoryList = ["Stack","Queue","Set","Dequeue","Priority Queue","List(Array)", "MultiSet","Dictionary"]
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: please remove

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        tableView.separatorColor = #colorLiteral(red: 0, green: 0, blue: 0, alpha: 1)
+        return dataStructureManipulatios.getAmountOfValuesInDataSorce()
+        
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "DataStructureCell", for: indexPath) as! DataStructureCell
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: how can we avoid force unwrap?

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+//
+//  Created by Alina Zaitseva on 4/12/18.
+//  Copyright © 2018 Alina Zaitseva. All rights reserved.
+//
+
+import UIKit
+
+class DataStructureTableViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    
+    let dataStructureManipulatios = DataCaprureMemoryStructModel()
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: Não falo português, could you use more "English" names?

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return dataStructureManipulatios.getAmountOfValuesInDataSorce()
+        
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "DataStructureCell", for: indexPath) as! DataStructureCell
+        
+        cell.titleLabel.text = dataStructureManipulatios.getValueInDataSource(item: indexPath.row)
+        
+//        cell.titleLabel.text = memoryList[indexPath.row]
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: please remove commented line

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return "Section \(section)"
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: Why do we do that?

File: DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemorySorce.swift [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+//
+//  Created by Alina Zaitseva on 4/13/18.
+//  Copyright © 2018 Alina Zaitseva. All rights reserved.
+//
+
+import Foundation
+
+protocol DataStructureMemorySorce {
+    
+    var listOfDataStructure: [String] { get }
```
Comment (AlexeyTyurenkov) [inline DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemorySorce.swift]: why do we need whole array there?

**Review by AlexeyTyurenkov — CHANGES_REQUESTED** [review]

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift (hunk 2 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: check if it is not nil

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift (hunk 3 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: Sure)

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift (hunk 5 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: we don't need it actually, I'll remove it

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift (hunk 4 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: okey

File: DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift (hunk 1 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/DataStructureTableViewController.swift]: okey

File: DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemorySorce.swift (hunk 6 above)
Comment (alinazaitseva1) [reply DataStructureMemories/DataStructureMemories/ProtocolDataStructure/DataStructMemorySorce.swift]: I'll remove

**Pull request merged**